src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/Logviewer.java

```diff
@@ -0,0 +1,250 @@
+    );
+  }
+
+  @Then("Click on log level toggle")
+  public void clickLogLevelToggle() {
+    WebElement element = Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId("log-viewer-log-level-toggle")
+    );
+    WaitHelper.waitForElementToBeDisplayed(element);
+    ElementHelper.clickOnElement(element);
+  }
+
+  @Then("Log level {string} should exist")
+  public void logLevelItemShouldExist(String item) {
+    Assert.assertEquals(true,
+      Helper.isElementExists(Helper.getCssSelectorByDataTestId("log-level-row-" + item))
+    );
+  }
+
+  @Then("Log level {string} checkmark should exist")
+  public void logLevelItemCheckShouldExist(String item) {
+    Assert.assertEquals(true,
+      Helper.isElementExists(Helper.getCssSelectorByDataTestId("log-level-check"),
+        Helper.locateElementByCssSelector(Helper.getCssSelectorByDataTestId("log-level-row-" + item))
+      )
+    );
+  }
+
+  @Then("Log level {string} checkmark should not exist")
+  public void logLevelItemCheckShouldNotExist(String item) {
+    Assert.assertEquals(false,
+      Helper.isElementExists(Helper.getCssSelectorByDataTestId("log-level-check"),
+        Helper.locateElementByCssSelector(Helper.getCssSelectorByDataTestId("log-level-row-" + item))
+      )
+    );
+  }
+
+  @Then("Log viewer content should contain message {string}")
+  public void logViewerContentMessageExists(String message) {
+    Assert.assertEquals(true,
+      Helper.isElementExists(By.xpath("//*[contains(text(), '" + message + "')]"),
+        Helper.locateElementByCssSelector(Helper.getCssSelectorByDataTestId("log-viewer-content"))
+      )
+    );
+  }
+
+  @Then("Log viewer content should not contain message {string}")
+  public void logViewerContentMessageNotExists(String message) {
+    Assert.assertEquals(false,
+      Helper.isElementExists(By.xpath("//*[contains(text(), '" + message + "')]"),
+        Helper.locateElementByCssSelector(Helper.getCssSelectorByDataTestId("log-viewer-content"))
+      )
+    );
+  }
+
+  @Then("Click on advanced logs")
+  public void clickAdvancedLogs() {
+    ElementHelper.clickOnElement(Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId("view-advanced-logs")
+    ));
+    Helper.waitSeconds();
+  }
+
+  @Then("Click on log level {string}")
+  public void clickLogLevelItem(String item) {
+    ElementHelper.clickOnElement(Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId("log-level-row-" + item)
+    ));
+    Helper.waitSeconds();
+  }
+
+  @Then("Log level popover should not show")
+  public void logLevelPopoverShouldHide() {
+    WaitHelper.waitForElementToBeHidden(Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId("log-viewer-log-level-popover")
+    ));
+  }
+
+  @Then("Scroll up to center")
+  public void scrollUpToCenter() {
+    JavascriptExecutor js = (JavascriptExecutor) SeleniumDriver.getDriver();
```

Comment:
I don't know how to scroll a container other than using jsexecutor

Reply:
Have you looked at the helper methods in here:
https://github.com/cdapio/cdap-e2e-tests/blob/727a948be8abdad2dd8d32900c606285195b0e30/src/main/java/io/cdap/e2e/utils/ElementHelper.java#L38

Reply:
Yea, the issue is the behavior here is not "scrolling to an element", it kinda just wants to scroll for a fixed length, cypress had the ability to scroll to 'center', 'bottom', but I don't think selenium can do that

Reply:
How do you decide what the fixed length is?

Reply:
experiments and estimations

Reply:
I did a quick search and found this:
```javascript
js.executeScript("arguments[0].scrollIntoView({behavior: "smooth", block: "center", inline: "nearest"}))", elem);
```

have you tried it?

Reply:
Let me try it out

Reply:
no this doesn't work